spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -113,9 +131,50 @@ public TaskDefinitionResource save(@RequestParam("name") String name,
 	}
 
 	/**
+	 * Register a composed task definition for future execution.
+	 *
+	 * @param name of the composed task
+	 * @param graph DSL definition for the composed task
+	 */
+	@RequestMapping(value = "/compose", method = RequestMethod.POST)
+	public TaskDefinitionResource saveComposedTask(@RequestParam("name") String name,
+			@RequestParam("definition") String graph) {
+		TaskDefinition taskDefinition = new TaskDefinition(name, graph, true);
+		TaskDefinition composedTaskDefinition = new TaskDefinition(
+				taskDefinition.getName(),getComposedArgs(taskDefinition.getDslText()));
+		validateComposedTaskElementsExist(name, graph);
+		repository.save(composedTaskDefinition);
+		return taskAssembler.toResource(taskDefinition);
+	}
+
+	private void validateComposedTaskElementsExist(String name, String graph) {
+		ComposedTaskValidator composedTaskValidator =
+				new ComposedTaskValidator(repository);
+		ComposedTaskNode composedTaskNode = new ComposedTaskParser().parse(name, graph);
+		composedTaskNode.accept(composedTaskValidator);
+		if(composedTaskValidator.getInvalidDefinitions().size() > 0) {
+			List<ComposedTaskValidationProblem> composedTaskValidationProblems =
+					new ArrayList<>();
+			composedTaskValidator.getInvalidDefinitions().keySet().stream().
+					forEach(s -> composedTaskValidationProblems.add(
+					new ComposedTaskValidationProblem(graph,
+							composedTaskValidator.getInvalidDefinitions().get(s),
+							DSLMessage.CT_ELEMENT_IN_COMPOSED_DEFINITION_DOES_NOT_EXIST))
+							);
+			throw new ComposedTaskValidationException(composedTaskNode,
+					composedTaskValidationProblems);
+		}
+	}
+
+	private String getComposedArgs(String graph) {
+		return String.format("%s --graph=\"%s\"", composedTaskProperties.getTaskName(), graph);
```

Comment:
Still trying to figure this out...    For example a user can embed quotes already.  i.e.
`task compose delme --graph "AAA '\"FAILED\"' -> CCC && BBB"` -or-
`task compose delme --graph "AAA ''FAILED'' -> CCC && BBB"`
this pattern follows what we already do here:
`stream create adf --definition "http --port=9700 | filter --expression=payload.matches('\"hello\"') | file"`